qiskit_experiments/curve_analysis/composite_curve_analysis.py

```diff
@@ -0,0 +1,452 @@
+from .curve_data import FitOptions, CurveData
+from .utils import analysis_result_to_repr, eval_with_uncertainties
+from .visualization import MplCurveDrawer, BaseCurveDrawer
+
+
+class CompositeCurveAnalysis(BaseCurveAnalysis):
+    r"""Composite Curve Analysis.
+
+    The :class:`.CompositeCurveAnalysis` takes multiple curve analysis instances
+    and performs each analysis on the same experimental results.
+    These analyses are performed independently, thus fit parameters have no correlation.
+    Note that this is different from :class:`.CompositeAnalysis` which assumes
+    analyzing the outcome of composite experiment, in which multiple different
+    experiments are performed.
+    The :class:`.CompositeCurveAnalysis` is attached to a single experiment instance,
+    which may execute similar circuits with slightly different settings.
+    Experiments with different settings might be distinguished by the circuit
+    metadata, and outcomes of the same set of experiments are assigned to a
+    specific analysis instance in the composite curve analysis.
+    This mapping is usually done with the analysis option ``filter_data`` dictionary.
+    Otherwise, all analyses are performed on the same set of outcomes.
+
+    Examples:
+
+        In this example, we write up a composite analysis consisting of two oscillation
+        analysis instances, assuming two Rabi experiments in 1-2 subspace
+        starting with different initial states :math:`\in \{|0\rangle, |1\rangle\}`.
+        This is a typical procedure to measure the thermal population of the qubit.
+
+        .. code-block:: python
+
+            from qiskit_experiments import curve_analysis as curve
+
+            analyses = []
+            for qi in (0, 1):
+                analysis = curve.OscillationAnalysis(name=f"init{qi}")
+                analysis.set_options(
+                    return_fit_parameters=["freq"],
+                    filter_data={"init_state": qi},
+                )
+            analysis = CompositeCurveAnalysis(analyses=analyses)
+
+        This ``analysis`` will return two analysis result data for the fit parameter "freq"
+        for experiments with the initial state :math:`|0\rangle` and :math:`|1\rangle`.
+        The experimental circuits starting with different initial states must be
+        distinguished by the circuit metadata ``{"init_state": 0}`` or ``{"init_state": 1}``,
+        along with the "xval" in the same dictionary.
+        If you want to compute another quantity using two fitting outcomes, you can
+        override :meth:`CompositeCurveAnalysis._create_curve_data` in subclass.
+
+    :class:`.CompositeCurveAnalysis` may override following methods.
+
+    .. rubric:: _evaluate_quality
+
+    This method evaluates the quality of the composite fit based on
+    the all analysis outcomes.
+    This returns "good" when all fit outcomes are evaluated as "good",
+    otherwise it returns "bad".
+
+    .. rubric:: _create_curve_data
```

Comment:
Should this be `_create_analysis_results`?

Reply:
Good catch! Thanks